internal/transport/transport_test.go

```diff
@@ -3204,3 +3206,141 @@ func (s) TestClientTransport_Handle1xxHeaders(t *testing.T) {
+		wantStreamSucceeded int64
+		wantStreamFailed    int64
+	}{
+		{
+			name:                "StreamsSucceeded",
+			eosReceived:         true,
+			removeFromActive:    false,
+			wantStreamSucceeded: 1,
+			wantStreamFailed:    0,
+		},
+		{
+			name:                "StreamsFailed",
+			eosReceived:         false,
+			removeFromActive:    false,
+			wantStreamSucceeded: 0,
+			wantStreamFailed:    1,
+		},
+		{
+			name:                "NonActiveStream",
+			eosReceived:         true,
+			removeFromActive:    true, // Stream already removed from activeStreams
+			wantStreamSucceeded: 0,
+			wantStreamFailed:    0,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			// Setup server configuration with channelz support
+			serverConfig := &ServerConfig{
+				ChannelzParent: channelz.RegisterServer("test server"),
+			}
+			defer channelz.RemoveEntry(serverConfig.ChannelzParent.ID)
+
+			// Create server and client with normal handler (not notifyCall)
+			server, client, cancel := setUpWithOptions(t, 0, serverConfig, normal, ConnectOptions{})
+			defer cancel()
+			defer server.stop()
+			defer client.Close(fmt.Errorf("test cleanup"))
+
+			// Wait for connection to be established
+			waitWhileTrue(t, func() (bool, error) {
+				server.mu.Lock()
+				defer server.mu.Unlock()
+				if len(server.conns) == 0 {
+					return true, fmt.Errorf("timed-out while waiting for connection")
+				}
+				return false, nil
+			})
+
+			// Get the server transport
+			server.mu.Lock()
+			var serverTransport *http2Server
+			for st := range server.conns {
+				serverTransport = st.(*http2Server)
+				break
+			}
+			server.mu.Unlock()
+
+			if serverTransport == nil {
+				t.Fatal("Server transport not found")
+			}
+
+			// Always create a real stream through the client
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+			defer cancel()
+			clientStream, err := client.NewStream(ctx, &CallHdr{})
+			if err != nil {
+				t.Fatalf("Failed to create stream: %v", err)
+			}
+
+			// Wait for the stream to be created on the server side
+			var serverStream *ServerStream
+			waitWhileTrue(t, func() (bool, error) {
+				serverTransport.mu.Lock()
+				defer serverTransport.mu.Unlock()
+				for _, v := range serverTransport.activeStreams {
+					if v.id == clientStream.id {
+						serverStream = v
+						return false, nil
+					}
+				}
+				return true, nil
+			})
+
+			if serverStream == nil {
+				t.Fatalf("Server stream not found for client stream ID %d", clientStream.id)
+			}
+
+			// For the NonActiveStream test, remove the stream from activeStreams
+			if test.removeFromActive {
+				serverTransport.mu.Lock()
+				delete(serverTransport.activeStreams, serverStream.id)
+				serverTransport.mu.Unlock()
+			}
+
+			initialSucceeded := serverTransport.channelz.SocketMetrics.StreamsSucceeded.Load()
+			initialFailed := serverTransport.channelz.SocketMetrics.StreamsFailed.Load()
```

Comment:
These values will always be 0, right? If so, we can assert they are 0 and assert that the absolute values of the metrics later instead of deltas. This would make it clearer for readers to see exactly what is the expected value of the metric.